integ/persistence_test.go

```diff
@@ -148,13 +155,31 @@ func doTestPersistenceWorkflow(
 			SearchAttributes: []iwfidl.SearchAttribute{
 				expectedDatetimeSearchAttribute,
 			},
+			DataAttributes: []iwfidl.KeyValue{
+				expectedDataAttribute,
+			},
 			WorkflowConfigOverride:   config,
 			UseMemoForDataAttributes: ptr.Any(useMemo),
 		},
 	}
 	_, httpResp, err := reqStart.WorkflowStartRequest(wfReq).Execute()
 	panicAtHttpError(err, httpResp)
 
+	// TODO: Fix the issue with running queryHandler before new workflow which Continues as New has started
```

Comment:
Looking into the code again, I now remember..
We intentionally set the query handler after the continueAsNew/dumpInternal activity. https://github.com/indeedeng/iwf/blob/main/service/interpreter/workflowImpl.go#L98

This is to ensure the correctness. If we set the query handler before that, the query handler could return empty data(since the loading hasn't completed), which will be incorrect response. 

So we would rather return server errors and let the client retry later.

Reply:
I think we should add a comment at https://github.com/indeedeng/iwf/blob/main/service/interpreter/workflowImpl.go#L98. to remind people why this is not before https://github.com/indeedeng/iwf/blob/main/service/interpreter/workflowImpl.go#L63

Reply:
For this test, we can add a comment to explain why a sleep is necessary